hazelcast/src/main/java/com/hazelcast/internal/nio/NodeIOService.java

```diff
@@ -73,74 +69,36 @@
         RestApiConfig restApiConfig = config.getNetworkConfig().getRestApiConfig();
-        boolean isRestConfigPresent = isAdvancedNetwork
-                ? config.getAdvancedNetworkConfig().getEndpointConfigs().get(EndpointQualifier.REST) != null
-                : restApiConfig != null;
-
-        if (isRestConfigPresent) {
-            // ensure the legacy Hazelcast group properties are not provided
-            ensurePropertyNotConfigured(properties, GroupProperty.REST_ENABLED);
-            ensurePropertyNotConfigured(properties, GroupProperty.HTTP_HEALTHCHECK_ENABLED);
-        }
-
-        if (isRestConfigPresent && isAdvancedNetwork) {
-            restApiConfig = new RestApiConfig();
-            restApiConfig.setEnabled(true);
-
-            RestServerEndpointConfig restServerEndpointConfig = config.getAdvancedNetworkConfig().getRestEndpointConfig();
-            restApiConfig.setEnabledGroups(restServerEndpointConfig.getEnabledGroups());
 
-        } else if (!isRestConfigPresent) {
-            restApiConfig = new RestApiConfig();
-            if (checkAndLogPropertyDeprecated(properties, GroupProperty.REST_ENABLED)) {
-                restApiConfig.setEnabled(true);
-                restApiConfig.enableAllGroups();
-            }
-            if (checkAndLogPropertyDeprecated(properties, GroupProperty.HTTP_HEALTHCHECK_ENABLED)) {
-                restApiConfig.setEnabled(true);
-                restApiConfig.enableGroups(RestEndpointGroup.HEALTH_CHECK);
-            }
+        if (isAdvancedNetwork && advancedNetworkConfig.getEndpointConfigs().get(EndpointQualifier.REST) != null) {
+            RestServerEndpointConfig restServerEndpointConfig = advancedNetworkConfig.getRestEndpointConfig();
+            restApiConfig.setEnabled(true).setEnabledGroups(restServerEndpointConfig.getEnabledGroups());
         }
+        restApiConfig.enableAllGroups();
 
         return restApiConfig;
     }
 
-    @SuppressWarnings("deprecation")
-    private static MemcacheProtocolConfig initMemcacheProtocolConfig(HazelcastProperties properties, Config config) {
-        boolean isAdvancedNetwork = config.getAdvancedNetworkConfig().isEnabled();
-        MemcacheProtocolConfig memcacheProtocolConfig = config.getNetworkConfig().getMemcacheProtocolConfig();
-        boolean isMemcacheConfigPresent = isAdvancedNetwork
-                ? config.getAdvancedNetworkConfig().getEndpointConfigs().get(MEMCACHE) != null
-                : memcacheProtocolConfig != null;
+    private static MemcacheProtocolConfig initMemcacheProtocolConfig(Config config) {
+        AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
+        boolean isAdvancedNetwork = advancedNetworkConfig.isEnabled();
 
-        if (isMemcacheConfigPresent) {
-            // ensure the legacy Hazelcast group properties are not provided
-            ensurePropertyNotConfigured(properties, GroupProperty.MEMCACHE_ENABLED);
+        if (isAdvancedNetwork) {
```

Comment:
In addition to `isAdvancedNetwork`, we also need to check `config.getAdvancedNetworkConfig().getEndpointConfigs().get(MEMCACHE) != null` to conclude that we should return an enabled memcache protocol config.